### PR TITLE
fix: add `ignoreDeprecations` flag for TS 5.x

### PR DESCRIPTION
**Pull request by dummdidumm**
Related to #8650

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

File: packages/kit/src/core/sync/write_tsconfig.js [hunk 1]
```diff
@@ -123,7 +130,10 @@ export function write_tsconfig(kit, cwd = process.cwd()) {
 					lib: ['esnext', 'DOM', 'DOM.Iterable'],
 					moduleResolution: 'node',
 					module: 'esnext',
-					target: 'esnext'
+					target: 'esnext',
+
+					// TODO(v2): use the new flag verbatimModuleSyntax instead (requires support by Vite/Esbuild)
+					ignoreDeprecations: Number(ts.version.split('.')[0]) >= 5 ? '5.0' : undefined
```
Comment (Rich-Harris) [inline packages/kit/src/core/sync/write_tsconfig.js:136]: could we ever end up here if typescript isn't installed (in which case `ts.version` would blow up)?

File: packages/kit/src/core/sync/write_tsconfig.js [hunk 2]
```diff
@@ -4,6 +4,13 @@ import colors from 'kleur';
 import { posixify } from '../../utils/filesystem.js';
 import { write_if_changed } from './utils.js';
 
+/** @type {import('typescript')} */
+// @ts-ignore
+let ts = undefined;
+try {
+	ts = (await import('typescript')).default;
```
Comment (Rich-Harris) [inline packages/kit/src/core/sync/write_tsconfig.js:11]: I think we might need to import relative to the cwd? We ran into this before with `adapter-auto` — it's possible, though rare, for `typescript` to be installed in a package while `@sveltejs/kit` is installed in a workspace root

File: packages/kit/src/core/sync/write_tsconfig.js (hunk 1 above)
Comment (dummdidumm) [reply packages/kit/src/core/sync/write_tsconfig.js:136]: good point, AFAIK tsconfig is always written, even without TS installed, so we should check that

File: packages/kit/src/core/sync/write_tsconfig.js (hunk 2 above)
Comment (dummdidumm) [reply packages/kit/src/core/sync/write_tsconfig.js:11]: We're doing the same in `write_types` and I heard no complaints so far. Still do it?

Comment (Rich-Harris) [reply packages/kit/src/core/sync/write_tsconfig.js:10]: ah, yeah that's probably fine in that case. though we may as well DRY it out

**Review by Rich-Harris — APPROVED** [review]

**Pull request merged**